zarr/v3/common.py

```diff
@@ -59,24 +53,8 @@ def _structure_chunk_key_encoding_metadata(d: Dict[str, Any], _t) -> ChunkKeyEnc
     )
 
     def _structure_codec_metadata(d: Dict[str, Any], _t=None) -> CodecMetadata:
```

Comment:
not for now but it would be nice if the codec metadata was a `TypedDict` so we could expect for keys like "name".

Reply:
`name` is really the only shared attribute among all codecs. Even `configuration` is optional. I don't think there'd be much value in a TypedDict. This piece of code is in the middle of the deserialization; lots of untyped magic happening here anyways :-)